addons/account/models/account_move.py

```diff
@@ -252,6 +252,11 @@ def _get_default_invoice_incoterm(self):
         check_company=True)
     reversal_move_id = fields.One2many('account.move', 'reversed_entry_id')
 
+    # ==== Hacky fields ====
+    # Technical fields to activate cache mechanism on aggregating computed fields to recompute
+    payment_ids = fields.One2many('account.payment', 'move_id', string='Inversed field for account.payment')
+    statement_line_ids = fields.One2many('account.bank.statement.line', 'move_id', string='Inversed field for account.bank.statement.line')
```

Comment:
BTW I just noticed that those many2one fields (`account.payment.move_id` and `account.bank.statement.line.move_id`) are **not** indexed.
Have you tried indexing them instead?

Reply:
Side note: indexing them also makes the one2many faster!

Reply:
I've just tried and it doesn't help. Below is time measure for creating 1000 account.move in batch.

DB:
* 18 K account.move
* 7 K account.payment
* 2.5 K account_bank_statement_line;

BEFORE:

CPU times: user 11.6 s, sys: 458 ms, total: 12 s
Wall time: 17.4 s


AFTER indexes:

CPU times: user 12.1 s, sys: 494 ms, total: 12.6 s
Wall time: 17.7 s
```diff
diff --git a/addons/account/models/account_bank_statement.py b/addons/account/models/account_bank_statement.py
index 203b4b2bb01..16bce55c5db 100644
--- a/addons/account/models/account_bank_statement.py
+++ b/addons/account/models/account_bank_statement.py
@@ -508,6 +508,7 @@ class AccountBankStatementLine(models.Model):
     move_id = fields.Many2one(
         comodel_name='account.move',
         string='Journal Entry', required=True, readonly=True, ondelete='cascade',
+        index=True,
         check_company=True)
     statement_id = fields.Many2one(
         comodel_name='account.bank.statement',
diff --git a/addons/account/models/account_payment.py b/addons/account/models/account_payment.py
index c56c7d97433..d6354c6324d 100644
--- a/addons/account/models/account_payment.py
+++ b/addons/account/models/account_payment.py
@@ -36,6 +36,7 @@ class AccountPayment(models.Model):
     move_id = fields.Many2one(
         comodel_name='account.move',
         string='Journal Entry', required=True, readonly=True, ondelete='cascade',
+        index=True,
         check_company=True)
 
     is_reconciled = fields.Boolean(string="Is Reconciled", store=True,
```

Reply:
Shouldn't `_inherits` fields be indexed by default?

Reply:
Hmm, inversed fields don't make this operation much faster either :thinking: 

CPU times: user 11.2 s, sys: 424 ms, total: 11.6 s
Wall time: 16.4 s